internal/keyvalue/leveldb.go

```diff
@@ -83,102 +97,116 @@ func NewLDBDatabase(file string, cache int, handles int) (*LDBDatabase, error) {
-	db.quitLock.Lock()
-	defer db.quitLock.Unlock()
+// NewIterator creates a binary-alphabetical iterator over the entire keyspace
+// contained within the leveldb database.
+func (db *LeveldbDatabase) NewIterator() database.Iterator {
+	return db.NewIteratorWithPrefix(nil)
+}
 
-	if db.quitChan != nil {
-		errc := make(chan error)
-		db.quitChan <- errc
-		if err := <-errc; err != nil {
-			db.log.Error("Metrics collection failed", "err", err)
-		}
-		db.quitChan = nil
-	}
-	err := db.db.Close()
-	if err == nil {
-		db.log.Info("Database closed")
-	} else {
-		db.log.Error("Failed to close database", "err", err)
-	}
+// NewIteratorWithPrefix creates a binary-alphabetical iterator over a subset
+// of database content with a particular key prefix.
+func (db *LeveldbDatabase) NewIteratorWithPrefix(prefix []byte) database.Iterator {
+	return db.db.NewIterator(util.BytesPrefix(prefix), nil)
 }
 
-func (db *LDBDatabase) LDB() *leveldb.DB {
-	return db.db
+// Stat returns a particular internal stat of the database.
+func (db *LeveldbDatabase) Stat(property string) (string, error) {
+	return db.db.GetProperty(property)
 }
 
-// Meter configures the database metrics collectors and
-func (db *LDBDatabase) Meter(prefix string) {
-	// Initialize all the metrics collector at the requested prefix
-	db.compTimeMeter = metrics.NewRegisteredMeter(prefix+"compact/time", nil)
-	db.compReadMeter = metrics.NewRegisteredMeter(prefix+"compact/input", nil)
-	db.compWriteMeter = metrics.NewRegisteredMeter(prefix+"compact/output", nil)
-	db.diskReadMeter = metrics.NewRegisteredMeter(prefix+"disk/read", nil)
-	db.diskWriteMeter = metrics.NewRegisteredMeter(prefix+"disk/write", nil)
-	db.writeDelayMeter = metrics.NewRegisteredMeter(prefix+"compact/writedelay/duration", nil)
-	db.writeDelayNMeter = metrics.NewRegisteredMeter(prefix+"compact/writedelay/counter", nil)
-
-	// Create a quit channel for the periodic collector and run it
-	db.quitLock.Lock()
-	db.quitChan = make(chan chan error)
-	db.quitLock.Unlock()
+// Compact flattens the underlying data store for the given key range. In essence,
+// deleted and overwritten versions are discarded, and the data is rearranged to
+// reduce the cost of operations needed to access them.
+//
+// A nil start is treated as a key before all keys in the data store; a nil limit
+// is treated as a key after all keys in the data store. If both is nil then it
+// will compact entire data store.
+func (db *LeveldbDatabase) Compact(start []byte, limit []byte) error {
+	return db.db.CompactRange(util.Range{Start: start, Limit: limit})
+}
 
-	go db.meter(3 * time.Second)
+// Path returns the path to the database directory.
+func (db *LeveldbDatabase) Path() string {
+	return db.fn
 }
 
 // meter periodically retrieves internal leveldb counters and reports them to
 // the metrics subsystem.
 //
-// This is how a stats table look like (currently):
+// This is how a stats leveldbTable look like (currently):
```

Comment:
Nit: This would read better as `// This is what a LevelDB stats table currently looks like:`

Reply:
I think this suffered a find/replace. Fixed it either way.